diff --git a/node_modules/circom_runtime/build/main.cjs b/node_modules/circom_runtime/build/main.cjs
index 235f1b6..3098349 100644
--- a/node_modules/circom_runtime/build/main.cjs
+++ b/node_modules/circom_runtime/build/main.cjs
@@ -77,7 +77,7 @@ async function builder(code, options) {
 
     options = options || {};
 
-    let memorySize = 32767;
+    let memorySize = 0;
     let memory;
     let memoryAllocated = false;
     while (!memoryAllocated){
diff --git a/node_modules/circom_runtime/js/witness_calculator.js b/node_modules/circom_runtime/js/witness_calculator.js
index 7db0bf8..d50bfec 100644
--- a/node_modules/circom_runtime/js/witness_calculator.js
+++ b/node_modules/circom_runtime/js/witness_calculator.js
@@ -24,7 +24,7 @@ export default async function builder(code, options) {
 
     options = options || {};
 
-    let memorySize = 32767;
+    let memorySize = 0;
     let memory;
     let memoryAllocated = false;
     while (!memoryAllocated){
