diff --git a/node_modules/circomlibjs/build/main.cjs b/node_modules/circomlibjs/build/main.cjs
index 41b3edb..e788e64 100644
--- a/node_modules/circomlibjs/build/main.cjs
+++ b/node_modules/circomlibjs/build/main.cjs
@@ -27388,3 +27388,9 @@ exports.mimc7Contract = mimc7Contract;
 exports.mimcSpongecontract = mimcSpongecontract;
 exports.newMemEmptyTrie = newMemEmptyTrie;
 exports.poseidonContract = poseidonContract;
+exports.BabyJub = BabyJub;
+exports.PedersenHash = PedersenHash;
+exports.Mimc7 = Mimc7;
+exports.MimcSponge = MimcSponge;
+exports.Eddsa = Eddsa;
+exports.getCurveFromName = ffjavascript.getCurveFromName;
\ No newline at end of file
